src/test/java/lambda/part3/exercise/ReachIterableTest.java

```diff
@@ -25,16 +28,14 @@
         default ReachIterable<T> filter(final Predicate<T> predicate) {
             return consumer -> {
                 boolean hasNext = true;
-                final boolean[] matched = { false };
+                final AtomicBoolean matched = new AtomicBoolean(false);
```

Comment:
I thought `Atomic*` wrapper is easier to read and work with than writing to array, isn't it? I understand that `volatile` can be expansive, but does it matter here?
I'd prefer to use simple mutable value-object for this purpose, but can't make it private in interface.